carma-messenger-core/cpp_message/src/MobilityRequest_Message.cpp

```diff
@@ -0,0 +1,538 @@
+/*
+ * Copyright (C) 2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+/**
+ * CPP File containing Mobility Request Message method implementations
+ */
+
+#include "MobilityRequest_Message.h"
+#include "MobilityHeader_Message.h"
+
+namespace cpp_message
+{
+    boost::optional<cav_msgs::MobilityRequest> Mobility_Request::decode_mobility_request_message(std::vector<uint8_t>& binary_array)
+    {
+        cav_msgs::MobilityHeader header;
+        cav_msgs::MobilityRequest output;
+
+        //decode results - stored in binary array
+        asn_dec_rval_t rval;
+        MessageFrame_t* message=nullptr;
+
+        //copy from vector to array
+        size_t len=binary_array.size();
+
+        uint8_t buf[len];
+        for(size_t i=0;i<len;i++){
```

Comment:
Kind of curious now that I saw it above, is there a reason to use these for-loops vs `std::copy()`? Ultimately they do the same thing, `std::copy()` is a bit cleaner looking/easier to read? Not saying you have to change it, but it's worth some consideration if you're comfortable with using std::copy().